network/ddclient-configuration.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+# shellcheck source=lib.sh
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+# Check if root
+root_check
+
+choice=$(whiptail --radiolist "This script lets you setup DynDNS by using the ddclient application.\nYou have to setup an account before you can start.\nPlease choose your DynDNS-Provider.\nSelect by pressing the spacebar and ENTER\n\nIf your DDNS-Provider isn't already supported, pleae open a new issue here:\n$ISSUES" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"Strato" "(strato.de)" OFF 3>&1 1>&2 2>&3)
+
+case "$choice" in
+    "Strato")
+        PROVIDER="Strato"
+        PROTOCOL="dyndns2"
+        SERVER="dyndns.strato.com"
+        USE_SSL="yes"
+    ;;
+    *)
+    ;;
+esac
+
+# Enter your Hostname
+while true
+do
+    HOSTNAME=$(whiptail --inputbox "Please enter the Host that you want to configure DDNS for. E.g. 'example.com'" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Is this correct? $HOSTNAME") ]]
+    then
+        print_text_in_color "$ICyan" "OK, please try again."
+        sleep 1
+    else
+        if [ -z "$HOSTNAME" ]
+        then
+            print_text_in_color "$ICyan" "Please don't leave the Inputbox empty."
+            sleep 1
+        else
+            break
+        fi
+    fi
+done
+
+# Enter your login
+while true
+do
+    LOGIN=$(whiptail --inputbox "Please enter the login for your DDNS provider. It will be most likely the Domain, that you want to setup. E.g. 'example.com'\nIf you are not sure, please refer to the documentation of your DDNS provider." "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Is this correct? $LOGIN") ]]
+    then
+        print_text_in_color "$ICyan" "OK, please try again."
```

Comment:
Add message box here please, I didn't think of the sleep before...

Thanks!

Reply:
fixed